aocd/models.py

```diff
@@ -10,11 +10,13 @@
 from functools import cache
 from functools import cached_property
 from importlib.metadata import entry_points
+from importlib.metadata import EntryPoint
 from itertools import count
 from pathlib import Path
 from textwrap import dedent
+from typing import TYPE_CHECKING, Callable, Generator, Optional, Protocol, TypeVar, TypedDict, cast, Iterable, Union, Literal
 
-from . import examples
+from . import examples as _examples # must rename import to avoid conflict w/ examples method
```

Comment:
Really? Even though examples is "namespaced" by the type (`Puzzle`) that it's defined on?  That's a bummer.

Reply:
Unfortunately, yes. Without doing this, `mypy` complains:

```
aocd/models.py:297: error: Name "examples.Example" is not defined  [name-defined]
```